### ISSUE TITLE
New Rule: enforce spacing around ternary operator (`?` and `:`)

### ISSUE THREAD

**Issue by dawsonhuang0** [eslint/eslint#19928]
### Rule details

Enforce consistent spacing around ternary `?` and `:` operators for readability.

### Related ECMAScript feature

Conditional (ternary) operator

### What type of rule is this?

Suggests an alternate way of doing something

### Example code

```js
// Bad (inconsistent spacing)
const result = condition?value1 :value2;

// Good (consistent spacing)
const result = condition ? value1 : value2;
```

### Why should this rule be in the core instead of a plugin?

This rule enforces consistent ternary formatting, which is a common readability concern in all codebases. Since ternary expressions are part of core JavaScript and used universally, this rule benefits from being in core ESLint rather than a plugin.

### Participation

- [x] I am willing to submit a pull request to implement this rule.

**Comment by Amnish04** [comment]
@dawsonhuang0 Thanks for opening the issue.

Spacing-related concerns like this are typically handled by dedicated formatters like [Prettier](https://prettier.io/), which already covers ternary formatting, including [advanced cases](https://prettier.io/blog/2023/11/13/curious-ternaries).

ESLint has also been moving away from enforcing formatting rules in core. As mentioned in [this ESLint blog post](https://eslint.org/blog/2023/10/deprecating-formatting-rules/#:~:text=In%20ESLint%20v8.,semicolons%2C%20string%20formats%2C%20etc.), formatting rules have been deprecated in favour of using tools like Prettier.

Maybe [@stylistic/eslint-plugin](https://github.com/eslint-stylistic/eslint-stylistic) would be the right project for this issue?

**Comment by dawsonhuang0** [comment]
Thanks for the context — that makes sense.

I wasn’t aware formatting rules were being phased out in ESLint core, but I totally understand the reasoning. Prettier is great at handling ternary formatting already, and it’s probably a better fit for this kind of rule.

I’ll explore contributing this to [@stylistic/eslint-plugin](https://github.com/eslint-stylistic/eslint-stylistic) instead. Appreciate the suggestion!
